File: app/src/main/java/org/tasks/data/CaldavTaskContainer.java
package org.tasks.data;

import androidx.room.Embedded;
import com.todoroo.astrid.data.Task;

public class CaldavTaskContainer {
  @Embedded public Task task;
  @Embedded public CaldavTask caldavTask;

  public Task getTask() {
    return task;
  }

  public CaldavTask getCaldavTask() {
    return caldavTask;
  }

  public String getRemoteId() {
    return caldavTask.getRemoteId();
  }

  public boolean isDeleted() {
    return task.isDeleted();
  }

  public String getVtodo() {
    return caldavTask.getVtodo();
  }

  @Override
  public String toString() {
    return "CaldavTaskContainer{" + "task=" + task + ", caldavTask=" + caldavTask + '}';
  }
}

File: app/src/main/java/org/tasks/data/CaldavTaskContainer.kt
package org.tasks.data

import androidx.room.Embedded
import com.todoroo.astrid.data.Task
import org.tasks.time.DateTime

class CaldavTaskContainer {
    @Embedded lateinit var task: Task
    @Embedded lateinit var caldavTask: CaldavTask

    val id: Long
        get() = task.id

    val remoteId: String?
        get() = caldavTask.remoteId

    val isDeleted: Boolean
        get() = task.isDeleted

    val sortOrder: Long
        get() = task.order ?: DateTime(task.creationDate).toAppleEpoch()

    val startDate: Long
        get() = task.hideUntil

    override fun toString(): String = "CaldavTaskContainer{task=$task, caldavTask=$caldavTask}"
}